opentracing-api/src/main/java/io/opentracing/io/opentracing/mocktracer/MockSpan.java

```diff
@@ -0,0 +1,219 @@
+package io.opentracing.io.opentracing.mocktracer;
+
+import io.opentracing.Span;
+import io.opentracing.SpanContext;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * MockSpans are created via MockTracer.buildSpan(...), but they are also returned via calls to
+ * MockTracer.finishedSpans(). They provide accessors to all Span state.
+ *
+ * @see MockTracer#finishedSpans()
+ */
+public class MockSpan implements Span {
```

Comment:
can you make the classes `final` please.
there's also a few fields through the PR that could also be final.